eqcorrscan/utils/correlate.py

```diff
@@ -24,18 +24,290 @@
 from __future__ import print_function
 from __future__ import unicode_literals
 
-import numpy as np
+import contextlib
 import ctypes
-from future.utils import native_str
+from multiprocessing import Pool as ProcessPool, cpu_count
+from multiprocessing.pool import ThreadPool
 
-from multiprocessing import Pool
+import numpy as np
+from future.utils import native_str
 from scipy.fftpack.helper import next_fast_len
+
 from eqcorrscan.utils.libnames import _load_cdll
 
+XCOR_FUNCS = {}  # cache of functions for doing cross correlations
+
+# methods added to each xcorr func registered
+# these implement the stream interface
+XCORR_STREAM_METHODS = ('multithread', 'multiprocess', 'concurrent',
+                        'stream_xcorr')
+
+# these implement the array interface
+XCOR_ARRAY_METHODS = ('array_xcorr')
+
+
+# ------------------ Context manager for switching out default
+class _Context:
+    """ class for permanently or temporarily changing items in a dict """
+
+    def __init__(self, cache, value_to_switch):
+        """
+        :param cache:
+        :param value_to_switch:
```

Comment:
Can we have some info for this doc-string?

Reply:
Sure. It should be noted this is a private class, so I don't really expect it to be initialized apart from the instance directly after the class definition, but it is good to document its behavior for other developers.

Reply:
Agreed on private-ness, more that I don't immediately understand what these args are, just a brief note on the the arguments are would help me out :)